src/main/java/timetable/TimeTableCommand.java

```diff
@@ -0,0 +1,212 @@
+            }
+        }
+        isInvalid = true;
+        String start;
+        String end;
+        int startTime = 0;
+        int endTime = 0;
+        while (isInvalid) {
+            System.out.println("Please enter the time of your activity (e.g. 6-9pm): ");
+            String time = in.nextLine();
+            try {
+                start = time.split("-")[0];
+                end = time.split("-")[1];
+                startTime = Integer.parseInt(start.replaceAll("[^0-9]", ""));
+                endTime = Integer.parseInt(end.replaceAll("[^0-9]", ""));
+                if (start.contains("am") || start.contains("pm")) {
+                    if (start.contains("pm") && startTime != 12) {
+                        startTime += 12;
+                    } else if (start.contains("am") && startTime == 12) {
+                        startTime = 0;
+                    }
+                    if (end.contains("pm") && startTime != 12) {
+                        endTime += 12;
+                    } else if (end.contains("am") && startTime == 12) {
+                        startTime = 0;
+                    }
+                } else if (end.contains("pm")) {
+                    endTime += 12;
+                    if (startTime != 12) {
+                        startTime += 12;
+                    }
+                }
+                if (startTime < endTime) {
+                    isInvalid = false;
+                }
+            } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+                System.out.println("You have entered an invalid time. Please try again");
+            }
+        }
+        LocalDateTime startDateTime = date.plusHours(startTime);
+        LocalDateTime endDateTime = date.withHour(endTime);
+        Duration duration = new Duration(startDateTime, endDateTime);
+        return new Activity(activityName, isOnline, linkOrVenue, duration);
+    }
+
+    public static void showLink(DateList dateList) {
+        LocalDate todayDate = LocalDateTime.now().toLocalDate();
+        int now = LocalDateTime.now().toLocalTime().getHour() * 100;
+        for (EventList eventList: dateList.dateList) {
+            if (eventList.dateTag.equals(todayDate)) {
+                accessEventList(eventList, todayDate, now);
+                return;
+            }
+        }
+    }
+
+    public static void accessEventList(EventList eventList, LocalDate todayDate, int now) {
+        for (Event event: eventList.events) {
+            for (Duration period: event.periods) {
+                if ((period.timeSlot.contains(now) || period.timeSlot.contains(now + 100)
+                        || period.timeSlot.contains(now + 200))
+                        && period.startDateTime.toLocalDate().equals(todayDate)) {
+                    System.out.print(event.linkOrVenue + "|" + event.name + "\n");
+                }
+            }
+        }
+    }
```

Comment:
Some parts of the code are still too deeply nested, and it would be good to apply better SLAP to it. But if it's unavoidable then just let it be i guess.